app/models/robots/bed/pooling_and_splitting.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Robots::Bed
+  # Pooling and Splitting beds can have multiple parents and multiple children,
+  # and have additional methods to support this
+  class PoolingAndSplitting < Robots::Bed::Base
+    attr_accessor :parents
+
```

Comment:
There's a bit of code duplication going on here, this is identical to the pooling bed in many places.